### PR TITLE
Emitter: Don't output trailing space for empty scalar nodes

### PR DESCRIPTION
**Pull request by perlpunk**
See issue #46

Passing emitter tests:
* 2XXW: Spec Example 2.25. Unordered Sets
* 5WE3: Spec Example 8.17. Explicit Block Mapping Entries
* 6KGN: Anchor for empty node
* 6XDY: Two document start markers
* 7W2P: Block Mapping with Missing Values
* 8KB6: Multiline plain flow mapping key without value
* 9BXH: Multiline doublequoted flow mapping key without value
* C2DT: Spec Example 7.18. Flow Mapping Adjacent Values
* JTV5: Block Mapping with Multiline Scalars
* KK5P: Various combinations of explicit block mappings
* LE5A: Spec Example 7.24. Flow Nodes
* UT92: Spec Example 9.4. Explicit Documents
* W42U: Spec Example 8.15. Block Sequence Entry Types
* W4TN: Spec Example 9.5. Directives Documents
* ZWK4: Key with anchor after missing explicit mapping value

### CODE REVIEW

File: src/emitter.c [hunk 1]
```diff
@@ -1923,7 +1923,14 @@ yaml_emitter_write_plain_scalar(yaml_emitter_t *emitter,
 
     STRING_ASSIGN(string, value, length);
 
-    if (!emitter->whitespace) {
+    if (!emitter->whitespace && (length || emitter->flow_level)) {
+        /**
+         * Avoid trailing spaces for empty values in block style.
+         * In flow style, the emitter forbids empty values everywhere,
+         * and outputs a single quoted scalar instead.
+         * If this is ever changed, checking for emitter->flow_level
```
Comment (ingydotnet) [inline src/emitter.c:1931]: I'm not sure I follow what you are saying here. Could you rephrase?

File: src/emitter.c [hunk 2]
```diff
@@ -1923,7 +1923,14 @@ yaml_emitter_write_plain_scalar(yaml_emitter_t *emitter,
...
+         * will prevent ambiguous things like {a:}
```
Comment (ingydotnet) [inline src/emitter.c:1932]: @perlpunk, how about...
```
If the emitter is ever changed to allow emitting empty values, then checking for emitter->flow_level
will prevent emitting ambiguous things like {a:}.
```
(with appropriate line wrapping of course :)

Comment (perlpunk) [reply src/emitter.c]: I rephrased it a bit more and pushed.

**Review by ShadowsatNight — APPROVED** [review]
src/emitter.c
